test/browser/browser_topsites_contextMenu_options.js

```diff
@@ -30,24 +30,46 @@ test_newtab({
+    // Don't count search topsites
+    let defaultTopSitesNumber = content.document.querySelectorAll(".top-site-outer:not(.placeholder):not(.search-shortcut)").length;
+    Assert.equal(defaultTopSitesNumber, 5, "5 top sites are loaded by default");
 
-    let secondTopSite = content.document.querySelector(".top-sites-list li:nth-child(2) a").getAttribute("href");
+    // Skip the search topsites select the second default topsite
+    let secondTopSite = content.document.querySelectorAll(".top-sites-list li:not(.search-shortcut):not(.placeholder)")[1].getAttribute("href");
 
-    let contextMenuItems = content.openContextMenuAndGetOptions(".top-sites-list li:first-child");
+    let contextMenuItems = content.openContextMenuAndGetOptions("li:not(.search-shortcut)");
     Assert.equal(contextMenuItems[4].textContent, "Dismiss", "'Dismiss' is the 5th item in the context menu list");
 
     contextMenuItems[4].querySelector("a").click();
 
-    // Need to wait for dismiss action.
-    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-sites-list li:first-child a").getAttribute("href") === secondTopSite,
-      "First topsite was dismissed");
+    // Wait for the topsite to be dismissed and the second one to replace it
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelector(".top-sites-list li:not(.search-shortcut):not(.placeholder)").getAttribute("href") === secondTopSite,
+      "First default topsite was dismissed");
 
-    defaultTopSitesNumber = content.document.querySelector(".top-sites-list").querySelectorAll("[class=\"top-site-outer\"]").length;
-    Assert.equal(defaultTopSitesNumber, 5, "5 top sites are displayed after one of them is dismissed");
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelectorAll(".top-site-outer:not(.placeholder):not(.search-shortcut)").length === 4, "4 top sites are displayed after one of them is dismissed");
   },
   async after() {
     await new Promise(resolve => NewTabUtils.undoAll(resolve));
   }
 });
+
+test_newtab({
+  before: setDefaultTopSites,
+  test: async function searchTopSites_dismiss() {
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelectorAll(".search-shortcut").length === 2,
+      "2 search topsites are loaded by default");
+
+    let contextMenuItems = content.openContextMenuAndGetOptions(".search-shortcut");
+    is(contextMenuItems.length, 2, "Search TopSites should only have Unpin and Dismiss");
+
+    // Unpin
+    contextMenuItems[0].querySelector("a").click();
+
+    await ContentTaskUtils.waitForCondition(() => content.document.querySelectorAll(".search-shortcut").length === 1,
+      "1 search topsite displayed after we unpin the other one");
+  },
+  after: () => {
+    // This is modified when calling .unpin in `disableSearchImprovementsPrefs`
+    Services.prefs.clearUserPref("browser.newtabpage.activity-stream.improvesearch.topSiteSearchShortcuts.havePinned");
```

Comment:
Is this necessary? At least the comment is outdated. Potentially we want to clean up / prepare for fresh top search more generally?

Reply:
The pref clear is required for multiple test runs in the same browser (`--verify`) which is what the TV mochitest suite (I think) is doing. On the second test run the TopSitesFeed wouldn't re-pin the unpinned topsite.
`disableSearchImprovements` would do a proper cleanup, you would need to toggle the `SEARCH_SHORCUTS_EXPERIMENT` in order to trigger that. I would do it as a follow up because it involves more code (wait & assert that cleanup happend).